File: problem-086/solution.go
package problem086

func minParanthesesRemove(s string) int {
	if len(s) < 2 {
		return len(s)
	}

	unclosed, invalid := 0, 0

	for _, c := range s {
		if c == '(' {
			unclosed++
		} else if unclosed == 0 {
			invalid++
		} else {
			unclosed--
		}
	}

	return invalid + unclosed
}

File: problem-086/solution.py
def min_parantheses_remove(parantheses: str) -> int:
    unclosed = invalid = 0
    for p in parantheses:
        if p == "(":
            unclosed += 1
        elif unclosed == 0:
            invalid += 1
        else:
            unclosed -= 1
    return unclosed + invalid
        